src/NuGet.Core/NuGet.Build.Tasks.Pack/PackTaskLogic.cs

```diff
@@ -447,6 +447,19 @@ private IEnumerable<OutputLibFile> InitLibFiles(IMSBuildItem[] libFiles)
                             string.Format(CultureInfo.CurrentCulture, Strings.InvalidPlatformVersion, targetFramework)
                         );
                     }
+                    var isNet5EraTfm = fw.Version.Major >= 5 &&
+                        StringComparer.OrdinalIgnoreCase.Equals(FrameworkConstants.FrameworkIdentifiers.NetCoreApp, fw.Framework);
+                    var isNetStandard = StringComparer.OrdinalIgnoreCase.Equals(FrameworkConstants.FrameworkIdentifiers.NetStandard, fw.Framework);
+                    if (isNet5EraTfm || isNetStandard)
+                    {
+                        var dotIdx = targetFramework.IndexOf('.');
+                        var dashIdx = targetFramework.IndexOf('-');
+                        var isDottedFwVersion = (dashIdx > -1 && dotIdx > -1 && dotIdx < dashIdx) || dotIdx > -1;
```

Comment:
```suggestion
                        var isDottedFwVersion = (dotIdx > -1 && dashIdx > dotIdx) || dotIdx > -1;
```
if `dotIdx > -1` and `dashIdx > dotIdx` then we can imply that `dashIdx > -1`. Hence removed that condition. Please help me understand why we need to check the presence of `-` char in the string.

Reply:
I see that this code is executed only when [packing via csproj, not nuspec](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Build.Tasks.Pack/PackTaskLogic.cs#L76). Don't we need this rule when packing using [`nuspec`](https://docs.microsoft.com/en-us/nuget/reference/msbuild-targets#packing-using-a-nuspec)?

Reply:
> Please help me understand why we need to check the presence of - char in the string.

We need to check for the dash because the dot might be in the _platform_ part of the tfm (`net50-windows7.0`)

> I see that this code is executed only when packing via csproj, not nuspec. Don't we need this rule when packing using nuspec?

I don't know why we do this, but I'm still figuring out how Pack works overall. @nkolev92 @zivkan do y'all have any idea what this difference is about?

Reply:
When we pack a nuspec, I don't believe that NuGet implicitly adds anything to the package. @zkat you should confirm, but I think it's 100% up to the customer writing he nuspec to explicitly define everything in the package.

However, this got me thinking, `TargetFramework` is now just an alias. NuGet should not be using this string in the `lib/<tfm>/` folder structure, it should be parsing the `TargetFrameworkMoniker` and `TargetPlatformMoniker` into a `NuGetFramework, then calling `.GetShortFolderName()`, which will correctly add the `.` in `net5.0`, even if it was missing in `TargetFramework`.

Maybe we need a test for this, but it shouldn't require a product change to pack.  I'd say maybe there should be a warning about this for restore, but it's not really feasible for NuGet to do the check, I think the .NET SDK would need to. By the time it gets to NuGet, we don't know if the TargetFramework was parsed, and therefore needs the warning, or if it was just an alias, and therefore we should not warn. Hence I think it's the .NET SDK's responsibility if this should be warned.

Reply:
I mostly agree with @zivkan.

We should add a test/warning for the manual nuspec scenario. Beyond that, it think the SDK might make more sense.